Graphics/Transform.cpp:167:26: warning: comparison between two arrays

Recent gcc compiler can be made to say:

CSFML-2.5.1/src/SFML/Graphics/Transform.cpp:167:26: warning: comparison between two arrays [-Warray-compare]

Source code is

   return (left->matrix == right->matrix) ? sfTrue : sfFalse;

--- a/src/SFML/Graphics/Transform.cpp
+++ b/src/SFML/Graphics/Transform.cpp
@@ -164,6 +164,9 @@ sfBool sfTransform_equal(sfTransform* left, sfTransform* right)
     CSFML_CHECK_RETURN(left, false);
     CSFML_CHECK_RETURN(right, false);
 
-    return (left->matrix == right->matrix) ? sfTrue : sfFalse;
+    for (int i = 0; i < 9; ++i) {
+        if (left->matrix[i] != left->matrix[i]) return sfFalse;
+    }
+    return sfTrue;
 }
 

